src/aal/aal_sparc.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#if defined(__arch64__) // More reliable than __sparc64__
+#  define SNMALLOC_VA_BITS_64
+#else
+#  define SNMALLOC_VA_BITS_32
+#endif
+
+namespace snmalloc
+{
+  /**
+   * Sparc architecture abstraction layer.
+   */
+  class AAL_Sparc
+  {
+  public:
+    /**
+     * Bitmap of AalFeature flags
+     */
+    static constexpr uint64_t aal_features = IntegerPointers;
+
+    static constexpr enum AalName aal_name = Sparc;
+
+#ifdef SNMALLOC_VA_BITS_64
+    /**
+     * Even Ultra-Sparc I supports 8192 and onwards
+     */
+    static constexpr size_t smallest_page_size = 0x2000;
+#else
+    static constexpr size_t smallest_page_size = 0x1000;
+#endif
+
+    /**
+     * On Sparc ideally pause instructions ought to be
+     * optimised per Sparc processor but here a version
+     * as least common denominator to avoid numerous ifdef,
+     * reading Conditions Code Register here
+     */
+    static inline void pause()
+    {
+      __asm__ volatile("rd %%ccr, %%g0" ::: "memory");
```

Comment:
It's probably worth at least mentioning here that as of OSA2011 there is a PAUSE ASR, intended to be used as:
```
wr %g0, reg_or_imm, %pause
```
where `reg_or_imm` is the number of cycles to pause for, and has:
```
pause reg_or_imm
```
as an assembly alias.

That's obviously not generally usable though for anything below v9e/v8pluse (E because it was the 5th revision to the V9 spec). What made you pick `rd %ccr` above anything else, out of interest? ASR 2 is not specified on V8 (see below) but maybe reading is required to not have any other architecturally-visible side-effects? The specification is not enormously clear.